common/src/main/java/tachyon/exception/TachyonException.java

```diff
@@ -65,11 +67,22 @@ public TachyonTException toTachyonTException() {
       if (e.getType() == throwInstance.getType()) {
         throw throwInstance;
       }
-      // TODO(andrew): Change this to something supported in Java 6
-    } catch (ReflectiveOperationException roe) {
-      // They passed us an exception class that couldn't be instantiated with a string and
-      // throwable, so we can ignore it
-      LOG.error("Class passed to unwrap is invalid: ", throwClass.getName());
+      // This will be easier when we move to Java 7, when instead of catching Exception we can catch
+      // ReflectiveOperationException, the Java 7 superclass for these Exceptions
+    } catch (InstantiationException e1) {
+      handleUnwrapExceptionError(throwClass);
+    } catch (IllegalAccessException e1) {
+      handleUnwrapExceptionError(throwClass);
+    } catch (InvocationTargetException e1) {
+      handleUnwrapExceptionError(throwClass);
+    } catch (NoSuchMethodException e1) {
+      handleUnwrapExceptionError(throwClass);
     }
   }
+
+  private static void handleUnwrapExceptionError(Class<?> throwClass) {
+    // They passed us an exception class that couldn't be instantiated with a string and
+    // throwable, so we can ignore it
+    LOG.error("Class passed to unwrap is invalid: ", throwClass.getName());
```

Comment:
I'm a little confused about why this is called unwrap when it wraps the passed in exception using the passed in class. If I have `te1` and I `unwrap(te1, te2)`, I'll throw a (`te2` caused by `te1`), where `te1` and `te2` have the same message. Should we be calling `newInstance(e.getMessage(), e.getCause())` instead?

Even crazier, could we just check `e.isInstance(throwClass)` and then cast to throwClass and throw if the exception is of type throwClass? The Java classes for our various TachyonExceptions should map 1:1 to the TachyonExceptionType enum, right?